src/main/java/com/kodilla/ecommercee/service/DbUserService.java

```diff
@@ -0,0 +1,71 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.domain.TokenNotFoundException;
+import com.kodilla.ecommercee.domain.User;
+import com.kodilla.ecommercee.domain.UserNotFoundException;
+import com.kodilla.ecommercee.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.security.SecureRandom;
+import java.util.List;
+import java.util.Optional;
+
+@Component
+public class DbUserService {
+    @Autowired
+    private UserRepository userRepository;
+
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    public Optional<User> getUser(final Long id) throws UserNotFoundException {
+        return userRepository.findById(id);
+    }
+
+    public void deleteUser(final Long id) {
+        if (userRepository.existsById(id)) {
+            userRepository.deleteById(id);
+        }
+    }
+
+    public User updateUser(final Long id, User user) throws UserNotFoundException {
+        User toUpdate = userRepository.findById(id).orElseThrow(UserNotFoundException::new);
+        toUpdate.setUsername(user.getUsername());
+        toUpdate.setToken(user.getToken());
+        toUpdate.setIsBlocked(user.getIsBlocked());
+        return userRepository.save(toUpdate);
+    }
+
+    public User saveUser(final User user) {
+        return userRepository.save(user);
+    }
+
+    public void blockUser(final Long id) {
+        if (userRepository.existsById(id)) {
+            User userToBlock = userRepository.getOne(id);
+            if (userToBlock.getIsBlocked().equals(false)) {
+                userToBlock.setIsBlocked(true);
+            } else {
+                userToBlock.setIsBlocked(false);
+            }
+        }
+    }
+
+    public void generateToken(final Long id) throws TokenNotFoundException {
+        String userTokenGeneration = "";
+        if (userRepository.existsById(id)) {
+            userTokenGeneration = tokenGeneration();
+            userRepository.getOne(id).setToken(userTokenGeneration);
+        }
+    }
```

Comment:
Żeby odłożyć wygenerowany token brakuje metody save.

Co do samego Tokena, zastanawiam się jeszcze jak ten mechanizm powinien działać. Czy ma ona pełnić zadanie przy autentykacji jak w Trello. Jak zobaczymy frontend to może coś się rozjaśni i rozpisze się nowe zadanie, ponieważ tam powinien być jeszcze mechanizm wygasania tokena  :)

Reply:
Właśnie wiem o tym i to był jedyny pomysł na jaki wpadłem "na ten czas". Ogólnie rzecz biorąc badałem kilka różnych opcji i ŻADNA (sic!) nie jest idealna. Co do zasady występują problemy na linii JSON - LocalDate i nie można tego zrobić z godziną, tylko w oparciu o czystą datę. To nas nie urządza, a wymaga tworzenia dodatkowego mappera.